node/src/store/header_ranges.rs

```diff
@@ -0,0 +1,530 @@
+use std::fmt::Display;
+use std::ops::RangeInclusive;
+use std::vec;
+
+use celestia_types::ExtendedHeader;
+use serde::Serialize;
+use smallvec::{IntoIter, SmallVec};
+
+use crate::store::utils::{ranges_intersection, try_consolidate_ranges, RangeScanResult};
+use crate::store::StoreError;
+
+pub type HeaderRange = RangeInclusive<u64>;
+
+macro_rules! header_ranges {
+    ( $( $r:expr ),* ) => {
+        {
+            HeaderRanges::from_vec(smallvec::smallvec![
+            $(
+                $r,
+            )*
+            ])
+        }
+    };
+}
+
+pub(crate) use header_ranges;
+
+/// Span of header that's been verified internally
+pub struct VerifiedHeaderSpan(Vec<ExtendedHeader>);
```

Comment:
I prefer `VerifiedExtentedHeaders` as a name

Reply:
I like it better too, changed 👍